### PR TITLE
Add/king validate functionality

### PR DESCRIPTION
**Pull request by TheraOzz**

### CODE REVIEW

File: app/models/pieces.py [hunk 1]
```diff
@@ -31,7 +32,28 @@ class Rook(Piece): ...
...
 
-class King(Piece): ...
+class King(Piece):
+    def validate_move(self, board: np.array, move: PieceMove):
+        super().validate_move(board, move)
+
+        # Restrict the king's movement to one block
+        if abs(move.from_pos[0] - move.to_pos[0]) <= 1 and abs(move.from_pos[1] - move.to_pos[1]) <= 1:
+            # Check if the move is in the same column
+            if move.from_pos[0] == move.to_pos[0]:
```
Comment (iandronis) [inline app/models/pieces.py:42]: There is the scenario where the "from" and the "to" position are the same.
In this case, we should raise a wrong move.

Comment (TheraOzz) [reply app/models/pieces.py:46]: nice one! resolved!

**Review by iandronis — APPROVED** [review]

File: app/models/pieces.py [hunk 2]
```diff
@@ -49,7 +51,28 @@ def validate_move(self, board: np.array, move: PieceMove):
...
+                print("valid column move")
+                return True
```
Comment (iandronis) [inline app/models/pieces.py:63]: Also you do not need to return True. Just make a return.

File: app/models/pieces.py [hunk 3]
```diff
@@ -49,7 +51,28 @@ def validate_move(self, board: np.array, move: PieceMove):
...
 
-class King(Piece): ...
+class King(Piece):
+    def validate_move(self, board: np.array, move: PieceMove):
+        super().validate_move(board, move)
+
+        # Restrict the king's movement to one block
+        if abs(move.from_pos[0] - move.to_pos[0]) <= 1 and abs(move.from_pos[1] - move.to_pos[1]) <= 1:
+            # Check if the move is in the same column
+            if move.from_pos[0] == move.to_pos[0]:
```
[the remark this answers is not in the document]
Comment (TheraOzz) [reply app/models/pieces.py:61]: lets do this later with a refactor once our strategy is clear, as you suggest in another comment.
good catch tho!

**Pull request merged**